Userland/Services/RequestServer/ConnectionFromClient.cpp

```diff
@@ -73,11 +75,31 @@ size_t ConnectionFromClient::on_header_received(void* buffer, size_t size, size_
     auto* request = static_cast<ActiveRequest*>(user_data);
     size_t total_size = size * nmemb;
     auto header_line = StringView { static_cast<char const*>(buffer), total_size };
+
+    // NOTE: We need to extract the HTTP reason phrase since it can be a custom value.
+    //       Fetching infrastructure needs this value for setting the status message.
+    if (!request->reason_phrase.has_value() && header_line.starts_with("HTTP/"sv)) {
+        if (auto const space_positions = header_line.find_all(" "sv); space_positions.size() > 1) {
+            auto const second_space_offset = space_positions.at(1);
+            auto const reason_phrase_string_view = header_line.substring_view(second_space_offset + 1).trim_whitespace();
+
+            if (!reason_phrase_string_view.is_empty()) {
+                // FIXME: Use Infra::isomorphic_decode() instead when that is merged.
+                auto decoder = TextCodec::decoder_for_exact_name("ISO-8859-1"sv);
+                VERIFY(decoder.has_value());
```

Comment:
Well, we don't really crash for "unparseable data" but only if we can't get a Latin1 decoder. That _should_ never fail.


IMO, it's worth the crash if a really weird bug is happening where we can't get a decoder instead of silently dropping the header and continuing.